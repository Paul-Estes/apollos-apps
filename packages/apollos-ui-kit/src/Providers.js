import { nest } from 'recompose';

import { ThemeProvider, ThemeManager } from './theme';
import { LayoutProvider } from './LayoutContext';

export default nest(ThemeProvider, ThemeManager, LayoutProvider);
