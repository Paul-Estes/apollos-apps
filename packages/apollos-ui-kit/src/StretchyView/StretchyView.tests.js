import React from 'react';
import renderer from 'react-test-renderer';

import { ScrollView, FlatList } from 'react-native';
import {
  ConnectedImage,
  GradientOverlayImage,
  PaddedView,
  H1,
  H2,
  H3,
} from '@apollosproject/ui-kit';
import Providers from '../Providers';

import StretchyView from '.';

const imageStyle = { width: '100%', aspectRatio: 1 };

describe('the StretchyView component', () => {
  it('should render as a background', () => {
    const tree = renderer.create(
      <Providers>
        <StretchyView
          StretchyComponent={
            <GradientOverlayImage
              overlayColor="white"
              overlayType="featured"
              source={{ uri: 'https://picsum.photos/600/600' }}
              style={imageStyle}
            />
          }
        >
          {(scrollViewProps) => (
            <ScrollView
              {...scrollViewProps}
              onEndReached={() => console.log('yay!')}
            >
              <PaddedView>
                <H1 padded>Well hello there!</H1>
                <H2 padded>
                  Hello there!
                  {'\n'}
                </H2>
                <H3 padded>Hello there!</H3>
                <H1 padded>Try to scroll me around.</H1>
              </PaddedView>
            </ScrollView>
          )}
        </StretchyView>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as a header', () => {
    const tree = renderer.create(
      <Providers>
        <StretchyView
          StretchyComponent={
            <ConnectedImage
              style={imageStyle}
              source={{ uri: 'https://picsum.photos/600/600' }}
            />
          }
        >
          {({ stretchyHeight, ...scrollViewProps }) => (
            <FlatList
              {...scrollViewProps}
              data={[1, 2, 3, 4, 5]}
              keyExtractor={(item) => item}
              onEndReached={() => console.warn('done')}
              renderItem={() => (
                <PaddedView>
                  <H1 padded>Well bye there!</H1>
                  <H2 padded>
                    Hello there!
                    {'\n'}
                  </H2>
                  <H3 padded>Hello there!</H3>
                  <H1 padded>Try to scroll me around.</H1>
                </PaddedView>
              )}
            />
          )}
        </StretchyView>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
