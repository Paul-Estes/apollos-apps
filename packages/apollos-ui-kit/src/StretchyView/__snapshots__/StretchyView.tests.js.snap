// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the StretchyView component should render as a background 1`] = `
<View
  StretchyComponent={
    <pure(Component)
      maintainAspectRatio={true}
      overlayColor="white"
      overlayType="featured"
      source={
        Object {
          "uri": "https://picsum.photos/600/600",
        }
      }
      style={
        Object {
          "aspectRatio": 1,
          "width": "100%",
        }
      }
    />
  }
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <RCTScrollView
    Stretchy={[Function]}
    onEndReached={[Function]}
    onScroll={[Function]}
    scrollEventThrottle={0.5}
  >
    <View>
      <View
        style={
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <Text
          padded={true}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "InterUI-Black",
              "fontSize": 43,
              "lineHeight": 49.45,
              "paddingVertical": 46.44,
            }
          }
        >
          Well hello there!
        </Text>
        <Text
          padded={true}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "InterUI-Black",
              "fontSize": 36,
              "lineHeight": 41.4,
              "paddingBottom": 20,
              "paddingTop": 24,
            }
          }
        >
          Hello there!
          

        </Text>
        <Text
          padded={true}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "InterUI-Black",
              "fontSize": 24,
              "lineHeight": 27.6,
              "paddingBottom": 17.28,
              "paddingTop": 25.92,
            }
          }
        >
          Hello there!
        </Text>
        <Text
          padded={true}
          style={
            Object {
              "color": "#303030",
              "fontFamily": "InterUI-Black",
              "fontSize": 43,
              "lineHeight": 49.45,
              "paddingVertical": 46.44,
            }
          }
        >
          Try to scroll me around.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`the StretchyView component should render as a header 1`] = `
<View
  StretchyComponent={
    <mapProps(getContext(mapProps(withPropsOnChange(mapProps(ConnectedImage)))))
      source={
        Object {
          "uri": "https://picsum.photos/600/600",
        }
      }
      style={
        Object {
          "aspectRatio": 1,
          "width": "100%",
        }
      }
    />
  }
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <RCTScrollView
    Stretchy={[Function]}
    data={
      Array [
        1,
        2,
        3,
        4,
        5,
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.5}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Well bye there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 36,
                "lineHeight": 41.4,
                "paddingBottom": 20,
                "paddingTop": 24,
              }
            }
          >
            Hello there!
            

          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
                "paddingBottom": 17.28,
                "paddingTop": 25.92,
              }
            }
          >
            Hello there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Try to scroll me around.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Well bye there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 36,
                "lineHeight": 41.4,
                "paddingBottom": 20,
                "paddingTop": 24,
              }
            }
          >
            Hello there!
            

          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
                "paddingBottom": 17.28,
                "paddingTop": 25.92,
              }
            }
          >
            Hello there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Try to scroll me around.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Well bye there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 36,
                "lineHeight": 41.4,
                "paddingBottom": 20,
                "paddingTop": 24,
              }
            }
          >
            Hello there!
            

          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
                "paddingBottom": 17.28,
                "paddingTop": 25.92,
              }
            }
          >
            Hello there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Try to scroll me around.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Well bye there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 36,
                "lineHeight": 41.4,
                "paddingBottom": 20,
                "paddingTop": 24,
              }
            }
          >
            Hello there!
            

          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
                "paddingBottom": 17.28,
                "paddingTop": 25.92,
              }
            }
          >
            Hello there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Try to scroll me around.
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Well bye there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 36,
                "lineHeight": 41.4,
                "paddingBottom": 20,
                "paddingTop": 24,
              }
            }
          >
            Hello there!
            

          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
                "paddingBottom": 17.28,
                "paddingTop": 25.92,
              }
            }
          >
            Hello there!
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Black",
                "fontSize": 43,
                "lineHeight": 49.45,
                "paddingVertical": 46.44,
              }
            }
          >
            Try to scroll me around.
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
